app/activeproject.cpp

```diff
@@ -567,3 +549,49 @@ void ActiveProject::setProjectRole( const QString &role )
     emit projectRoleChanged();
   }
 }
+
+bool ActiveProject::recordingAllowed( QgsMapLayer *layer ) const
+{
+  if ( !layer )
+    return false;
+
+  if ( layer->readOnly() )
+    return false;
```

Comment:
Can you add a comment that this is only needed because of a bug in `QgsMapLayerProxyModel::layerMatchesFilters`?
Having `Qgis::LayerFilter::WritableLayer` should normally be enough